gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java

```diff
@@ -0,0 +1,250 @@
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.TaskProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateManifestTask extends DefaultTask {
+    private static final Logger log = LoggerFactory.getLogger(CreateManifestTask.class);
+    public static final String SLS_RECOMMENDED_PRODUCT_DEPS_KEY = "Sls-Recommended-Product-Dependencies";
+    public static final ObjectMapper jsonMapper = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setPropertyNamingStrategy(PropertyNamingStrategy.KEBAB_CASE);
+
+    private Property<String> serviceName = getProject().getObjects().property(String.class);
+    private Property<String> serviceGroup = getProject().getObjects().property(String.class);
+    private Property<ProductType> productType = getProject().getObjects().property(ProductType.class);
+
+    private MapProperty<String, Object> manifestExtensions = getProject().getObjects()
+            .mapProperty(String.class, Object.class);
+    private ListProperty<ProductDependency> productDependencies = getProject().getObjects()
+            .listProperty(ProductDependency.class);
+    private SetProperty<ProductId> ignoredProductIds = getProject().getObjects().setProperty(ProductId.class).empty();
+
+    private RegularFileProperty manifestFile = getProject().getObjects().fileProperty();
+    private Configuration productDependenciesConfig;
+
+    @Input
+    public final Property<String> getServiceName() {
+        return serviceName;
+    }
+
+    @Input
+    public final Property<String> getServiceGroup() {
+        return serviceGroup;
+    }
+
+    @Input
+    public final Property<ProductType> getProductType() {
+        return productType;
+    }
+
+    @Input
+    public final MapProperty<String, Object> getManifestExtensions() {
+        return manifestExtensions;
+    }
+
+    @Input
+    public final ListProperty<ProductDependency> getProductDependencies() {
+        return productDependencies;
+    }
+
+    @Input
+    public final SetProperty<ProductId> getIgnoredProductIds() {
+        return ignoredProductIds;
+    }
+
+    @InputFiles
+    public final FileCollection getProductDependenciesConfig() {
+        return productDependenciesConfig;
+    }
+
+    public final void setProductDependenciesConfig(Configuration productDependenciesConfig) {
+        this.productDependenciesConfig = productDependenciesConfig;
+    }
+
+    @Input
+    final String getProjectVersion() {
+        String stringVersion = String.valueOf(getProject().getVersion());
+        if (!SlsProductVersions.isValidVersion(stringVersion)) {
```

Comment:
I know all this validation logic lived in the `getProjectVersion` method before, but it feels a bit weird because Gradle might well run this method to determine whether the task is up to date or not and I think the failures are probably just annoying in that case.

Could we move it to a `validateProjectVersion` method or something that only gets run in the actual @TaskAction?